mitodl/micromasters
# Added coupon APIs

#### What are the relevant tickets?
Fixes #2237 

#### What's this PR do?
Adds REST APIs for getting information about coupons and to attach a user to a coupon.

#### How should this be manually tested?
Create a coupon using the Django shell using the `CouponFactory`. Go to `/api/v0/coupons/`. You should see an empty list. Add the coupon to your user using `UserCoupon.objects.create(coupon=coupon, user=user)` in the shell. Look at `/api/v0/coupons/` again and you should see one coupon.

In the shell delete the `UserCoupon` by running `UserCoupon.objects.filter(user=user, coupon=coupon).delete()`. Go to `/api/v0/coupons/<code>/users/` and POST to that API with `{"username": "your-username-here"}`. In the shell verify that the `UserCoupon` was created via `UserCoupon.objects.filter(user=user, coupon=coupon).exists()`
